### PR TITLE
[INTERNAL] Upgrade to use pip-compile

### PR DESCRIPTION
**Pull request by kfdm**
This PR is not concerned with updating any version requirements yet. For this PR I *only* want to cleanup our handling of requirements, and my next PR will actually look at updating the requirements themselves.

### CODE REVIEW

**Comment by kfdm** [comment]
Need to rebase it on the earlier dependabot changes

File: Makefile [hunk 1]
```diff
@@ -20,41 +20,41 @@ TARGET_MAX_CHAR_NUM=20
...
+			printf "  $(YELLOW)%-$(TARGET_MAX_CHAR_NUM)s$(RESET) $(GREEN)%s$(RESET)\n", helpCommand, helpMessage; \
 		} \
 	} \
 	{ lastLine = $$0 }' $(MAKEFILE_LIST)
 
-${PIP_BIN}:
-	python3 -m venv .venv
-	${PIP_BIN} install --upgrade pip setuptools wheel
+$(PIP_BIN):
+	python3.9 -m venv .venv
```
Comment (vincent-olivert-riera) [inline Makefile]: Why are we doing this change?

Comment (kfdm) [reply Makefile]: I think that's a change that leaked in. I'm going to try to get back to fixing this PR today.

Comment (kfdm) [reply Makefile]: Let me revert it to Draft PR again 🙇

Comment (vincent-olivert-riera) [reply Makefile]: Sure, no problem 👍🏻

**Comment by kfdm** [comment]
I believe I cleaned up the weirdness in the commits now.

**Pull request merged**